Fix link and remove unreferenced links in Android build page (#9422)
flutter/website: src/deployment/android.md

io/flutter/embedding/engine/loader/FlutterLoader.class
io/flutter/util/PathUtils.class
```

Also, include any other classes used in app startup.
For more detailed guidance on adding multidex support manually,
check out the official [Android documentation][multidex-docs].

## Reviewing the app manifest

Review the default [App Manifest][manifest] file, `AndroidManifest.xml`.
This file is located in `[project]/android/app/src/main`.

> Verify that the following values:

Reviewer: ```suggestion
Verify that the following values are correct:
```

Author: That would be [redundant](https://www.merriam-webster.com/dictionary/verify).

Reviewer: But "that" isn't explained in "Verify that the following values:". Perhaps "Verify the following values"?

> : Specify the final, unique [application ID][]

Reviewer: While you're here, consider adding punctuation to this definition entry to be consistent with the others.

  a reference to a string resource.

`buildToolsVersion`

> : If you're using [Android plugin for Gradle][] 3.0.0 or higher,

Reviewer: Since you're here, consider dropping the "If ..." part of this sentence. All versions of Android Studio that we support [require 3.2 or greater](https://developer.android.com/build/releases/gradle-plugin#android_gradle_plugin_and_android_studio_compatibility).

> : If you're using [Android plugin for Gradle][] 3.0.0 or higher,
>   your project automatically uses the default version of the
>   build tools that the plugin specifies. Alternatively,
>   you can specify a version of the build tools.

Author: **suggestion:** 
```suggestion
: The Gradle plugin specifies the default version of the
  build tools that your project uses.
  You can use this option to specify a different version of the build tools.
```

### Are there any special considerations with add-to-app?
{% endcomment %}

> [Android plugin for Gradle]: {{site.android-dev}}/studio/releases/gradle-plugin

Reviewer: Did you intentionally want this to link to the release notes? If not, https://developer.android.com/build seems like a better target for the text "Android plugin for Gradle".

[armeabi-v7a]: {{site.android-dev}}/ndk/guides/abis#v7a
[bundle]: {{site.android-dev}}/guide/app-bundle
[configuration qualifiers]: {{site.android-dev}}/guide/topics/resources/providing-resources#AlternativeResources
[fat APK]: https://en.wikipedia.org/wiki/Fat_binary
[flutter_launcher_icons]: {{site.pub}}/packages/flutter_launcher_icons
[Getting Started guide for Android]: {{site.material}}/develop/android/mdc-android
[GitHub repository]: {{site.github}}/google/bundletool/releases/latest
[Google Maven]: https://maven.google.com/web/index.html#com.google.android.material:material
[gradlebuild]: {{site.android-dev}}/studio/build/#module-level
[internal version number]: {{site.android-dev}}/studio/publish/versioning
[launchericons]: {{site.material}}/styles/icons
[manifest]: {{site.android-dev}}/guide/topics/manifest/manifest-intro

> [minimum API level]: {{site.android-dev}}/studio/publish/versioning#minsdkversion

Reviewer: Looks like this fragment doesn't exist on the linked page. Perhaps 

```suggestion
[minimum API level]: {{site.android-dev}}/studio/publish/versioning#minsdk
```
